ppl/src/main/java/com/amazon/opendistroforelasticsearch/sql/ppl/parser/AstExpressionBuilder.java

```diff
@@ -0,0 +1,177 @@
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.sql.ppl.parser;
+
+import com.amazon.opendistroforelasticsearch.sql.ppl.antlr.parser.OpenDistroPPLParser;
+import com.amazon.opendistroforelasticsearch.sql.ppl.antlr.parser.OpenDistroPPLParserBaseVisitor;
+import com.amazon.opendistroforelasticsearch.sql.ppl.plans.expression.AggregateFunction;
+import com.amazon.opendistroforelasticsearch.sql.ppl.plans.expression.And;
+import com.amazon.opendistroforelasticsearch.sql.ppl.plans.expression.AttributeList;
+import com.amazon.opendistroforelasticsearch.sql.ppl.plans.expression.DataType;
+import com.amazon.opendistroforelasticsearch.sql.ppl.plans.expression.EqualTo;
+import com.amazon.opendistroforelasticsearch.sql.ppl.plans.expression.Expression;
+import com.amazon.opendistroforelasticsearch.sql.ppl.plans.expression.Function;
+import com.amazon.opendistroforelasticsearch.sql.ppl.plans.expression.In;
+import com.amazon.opendistroforelasticsearch.sql.ppl.plans.expression.Literal;
+import com.amazon.opendistroforelasticsearch.sql.ppl.plans.expression.Not;
+import com.amazon.opendistroforelasticsearch.sql.ppl.plans.expression.Or;
+import com.amazon.opendistroforelasticsearch.sql.ppl.plans.expression.UnresolvedAttribute;
+import java.util.stream.Collectors;
+
+public class AstExpressionBuilder extends OpenDistroPPLParserBaseVisitor<Expression> {
+    /** Logical expression excluding boolean, eval, comparison */
+    @Override
+    public Expression visitLogicalNot(OpenDistroPPLParser.LogicalNotContext ctx) {
+        return new Not(visit(ctx.logicalExpression()));
+    }
+
+    @Override
+    public Expression visitLogicalOr(OpenDistroPPLParser.LogicalOrContext ctx) {
+        return new Or(visit(ctx.left), visit(ctx.right));
+    }
+
+    @Override
+    public Expression visitLogicalAnd(OpenDistroPPLParser.LogicalAndContext ctx) {
+        return new And(visit(ctx.left), visit(ctx.right));
+    }
+
+
+    /** Eval expression */
+    @Override
+    public Expression visitEvalExpression(OpenDistroPPLParser.EvalExpressionContext ctx) {
+        Expression field = visit(ctx.fieldExpression());
+        Expression evalFunctionCall = visit(ctx.evalFunctionCall());
+        return new EqualTo(field, evalFunctionCall);
+    }
+
+    /** Comparison expression */
+    @Override
+    public Expression visitCompareExpr(OpenDistroPPLParser.CompareExprContext ctx) {
+        Expression field = visit(ctx.left);
+        Expression value = visit(ctx.right);
+        String operator = ctx.comparisonOperator().getText();
+        switch (operator.replaceAll("\\s","")) {
```

Comment:
Is this necessary? I assume parser has handled this?

Reply:
Yeah makes sense, I overthought the cases. I will remove it. Thanks!